src/mui/list/Datagrid.js

```diff
@@ -109,39 +80,40 @@ class Datagrid extends Component {
                 style={options && options.fixedHeader ? null : styles.table}
                 fixedHeader={false}
                 {...options}
-              >
+            >
                 <TableHeader
                     displaySelectAll={false}
                     adjustForCheckbox={false}
                     {...headerOptions}
                 >
-                    <TableRow style={muiTheme.tableRow}>
-                        {React.Children.map(children, (field, index) => {
-                            if (childrenFilter(resource, field)) {
-                                return field ? (
-                                    <DatagridHeaderCell
-                                        key={field.props.source || index}
-                                        field={field}
-                                        defaultStyle={
-                                            index === 0 ? (
-                                                styles.header['th:first-child']
-                                            ) : (
-                                                styles.header.th
-                                            )
-                                        }
-                                        currentSort={currentSort}
-                                        isSorting={
-                                            field.props.source ===
-                                            currentSort.field
-                                        }
-                                        updateSort={this.updateSort}
-                                        resource={resource}
-                                    />
-                                ) : null;
-                            }
-                            return;
-                        })}
-                    </TableRow>
+                  <TableRow style={muiTheme.tableRow}>
```

Comment:
Be careful about the indent, aor uses 4 spaces indent, so when you auto indent it adds useless diff.